realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -663,7 +663,7 @@ void BaseRealSenseNode::publishAlignedDepthToOthers(rs2::frameset frames, const
         if (RS2_STREAM_DEPTH == stream_type)
             continue;
 
-        auto stream_index = frame.get_profile().stream_index();
+        auto stream_index = (stream_type==RS2_STREAM_INFRARED)?1:frame.get_profile().stream_index();
```

Comment:
Why is this required?  What if the stream type isn't INFRARED?

Reply:
Since infrared needs the index to match [this](https://github.com/plusone-robotics/realsense/blob/por_upstream_sync/realsense2_camera/include/realsense_node_factory.h#L29) which specifies it as a `1`. However when this is indexed normally using the `frame.get_profile().stream_index()` it returns `0`.

I realize this is a hack for now, and will be fixed soon